app/src/main/java/org/mozilla/fenix/home/sessioncontrol/viewholders/NoCollectionsMessageViewHolder.kt

```diff
@@ -5,59 +5,59 @@
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.ComposeView
+import androidx.compose.ui.unit.dp
 import androidx.lifecycle.LifecycleOwner
-import kotlinx.coroutines.ExperimentalCoroutinesApi
-import kotlinx.coroutines.flow.collect
-import kotlinx.coroutines.flow.map
 import mozilla.components.browser.state.selector.normalTabs
 import mozilla.components.browser.state.store.BrowserStore
-import mozilla.components.lib.state.ext.flowScoped
-import mozilla.components.support.ktx.kotlinx.coroutines.flow.ifChanged
+import mozilla.components.lib.state.ext.observeAsComposableState
 import org.mozilla.fenix.R
-import org.mozilla.fenix.databinding.NoCollectionsMessageBinding
-import org.mozilla.fenix.ext.increaseTapArea
+import org.mozilla.fenix.compose.CollectionsPlaceholder
+import org.mozilla.fenix.compose.ComposeViewHolder
 import org.mozilla.fenix.home.sessioncontrol.CollectionInteractor
-import org.mozilla.fenix.utils.view.ViewHolder
 
-@OptIn(ExperimentalCoroutinesApi::class)
-open class NoCollectionsMessageViewHolder(
-    view: View,
+/**
+ * [RecyclerView.ComposeViewHolder] for displaying a message detailing the collections feature and
+ * allowing users to easily start creating their first.
+ *
+ * @param composeView [ComposeView] which will be populated with Jetpack Compose UI content.
+ * @param viewLifecycleOwner [LifecycleOwner] to which this Composable will be tied to.
+ * @param store to receive live updates to normal tabs.
+ * @param interactor [CollectionInteractor] callback for user interaction.
+ */
+class NoCollectionsMessageViewHolder(
+    composeView: ComposeView,
     viewLifecycleOwner: LifecycleOwner,
-    store: BrowserStore,
-    interactor: CollectionInteractor
-) : ViewHolder(view) {
+    private val store: BrowserStore,
+    private val interactor: CollectionInteractor
+) : ComposeViewHolder(composeView, viewLifecycleOwner) {
 
     init {
-        val binding = NoCollectionsMessageBinding.bind(view)
-
-        binding.addTabsToCollectionsButton.apply {
-
-            setOnClickListener {
-                interactor.onAddTabsToCollectionTapped()
-            }
-            isVisible = store.state.normalTabs.isNotEmpty()
-        }
-
-        binding.removeCollectionPlaceholder.apply {
-            increaseTapArea(
-                view.resources.getDimensionPixelSize(R.dimen.tap_increase_16)
-            )
-            setOnClickListener {
-                interactor.onRemoveCollectionsPlaceholder()
-            }
-        }
-
-        store.flowScoped(viewLifecycleOwner) { flow ->
-            flow.map { state -> state.normalTabs.size }
-                .ifChanged()
-                .collect { tabs ->
-                    binding.addTabsToCollectionsButton.isVisible = tabs > 0
-                }
-        }
+        val horizontalPadding =
+            composeView.resources.getDimensionPixelSize(R.dimen.home_item_horizontal_margin)
+        val verticalPadding =
```

Comment:
I don't think we need this. I don't see any vertical padding that isn't already handled with your `Spacer(Modifier.height(40.dp))` in https://github.com/mozilla-mobile/fenix/blob/44fa89915467745af4ebf20fde1ab2eddac15411/app/src/main/res/layout/no_collections_message.xml#L11